src/H5private.h

```diff
@@ -49,29 +34,28 @@
 #include <limits.h>
 #include <math.h>
 #include <signal.h>
+#include <stdint.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
 
-/*
- * If _POSIX_VERSION is defined in unistd.h then this system is Posix.1
- * compliant. Otherwise all bets are off.
- */
+/* POSIX headers */
 #ifdef H5_HAVE_UNISTD_H
-#include <sys/types.h>
+#include <pwd.h>
 #include <unistd.h>
-#endif
-#ifdef _POSIX_VERSION
+#include <sys/types.h>
 #include <sys/wait.h>
-#include <pwd.h>
 #endif
 
-#include <stdint.h>
+/* Include the Pthreads header, if necessary */
+#if defined(H5_HAVE_THREADSAFE) && defined(H5_HAVE_PTHREAD_H)
+#include <pthread.h>
+#endif
```

Comment:
If you are on Windows and told us about Pthreads, we should include this header. There's no need for ifdef shenanigans.